swarm/network/simulation/events_test.go

```diff
@@ -90,14 +91,16 @@ func TestPeerEventsTimeout(t *testing.T) {
 				close(done)
 				return
 			} else {
-				t.Fatal(e.Error)
+				errC <- e.Error
 			}
 		}
 	}()
 
 	select {
 	case <-time.After(time.Second):
 		t.Error("no context deadline received")
```

Comment:
Should  `case <-time.After(time.Second)` contain `t.Fatal()` and `case err := <-errC` just `t.Error`.
As we might want to continue if we saw an error so that we can collect all errors and we want to terminate the test `time.After`.

Reply:
In this test function, there is no difference between t.Fatal and t.Error in this select case as it is the last place where errors are reported. There is nowhere to continue, and I see no possible changes to this test case that would extend the function after select case in the future.

Reply:
Then I would prefer to have either `t.Error` or `t.Fatal` and not both. As that would remove the cognitive load of deciding whether the distinction is for a reason or not.

Reply:
Then I would prefer to have either `t.Error` or `t.Fatal` and not both. As that would remove the cognitive load of deciding whether the distinction is for a reason or not.

Reply:
That is ok, updated.